src/controllers/csi/provisioner/controller.go

```diff
@@ -76,6 +83,34 @@ func (provisioner *OneAgentProvisioner) SetupWithManager(mgr ctrl.Manager) error
 }
 
 func (provisioner *OneAgentProvisioner) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	fs := afero.NewOsFs()
+	if provisioner.parallelDownloadsEnabled() {
+		if provisioner.parallelDownloadsLimitReached() {
+			log.Info("max parallel downloads reached, requeued")
+			return reconcile.Result{RequeueAfter: shortRequeueDuration}, nil
+		}
+		atomic.AddInt64(&provisioner.currentParallelDownloads, 1)
+		log.Info("staring parallel download")
+		go func() {
```

Comment:
What if download is still in progress and another reconciliation started?
Add `sync.WorkGroup` to wait for downloads to be finished before exiting reconciliation loop.

Reply:
that is not how it works 😅

Reply:
we would need a sync group per dynakube, and not block but requeue them

Reply:
If I understand it correctly, `provisioner.currentParallelDownloads` is a state shared between reconciler loops, so it may happen that two downloads for single tenant are in progress :thinking:.

Reply:
`Reconcile()` should not exit until downloads are done.